### DISCUSSION TITLE
running multiple scripts at the same time

### DISCUSSION THREAD

**Discussion by wif70068** [seleniumbase/SeleniumBase#2773] [Q&A] [answered]
if I run multiple python scripts which look like this:

```
...
with SB(uc=True) as sb:
    sb.driver.set_page_load_timeout(60)
    sb.driver.uc_open_with_reconnect(url, reconnect_time=10)
    sb.sleep(1)
    if sb.is_element_visible('iframe[src*="challenge"]'):
        with sb.frame_switch('iframe[src*="challenge"]'):
            sb.driver.uc_click("span.mark", reconnect_time=3)
            sb.sleep(5)
```
            
... at times when their execution overlaps, will the other running scripts be aborted by the implicit driver.quit() of the "with SB" of the first script finishing?

**Comment by mdmintz** [comment, marked as answer]
Your `"span.mark"` selector is incorrect. In case you missed the memo, Cloudflare changed their CAPTCHAs: No more `span.mark`. Only `span` now for that selector. (https://github.com/seleniumbase/SeleniumBase/issues/2766)

The `SB()` format will automatically quit the browser at the end of the `with` block.

For multithreading, you have two options:
1. `pytest` via `pytest-xdist`. (Eg. `pytest -n3`, as seen in [SeleniumBase/examples/presenter/multi_uc.py](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/presenter/multi_uc.py))
2. `ThreadPoolExecutor` as shown in [SeleniumBase/help_docs/uc_mode.md](https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/uc_mode.md).

**Comment by wif70068** [comment]
Thanks for the quick response and the selector update!

The suggested options seem ideal for one script managing multiple URLs/threads. My goal, however, is to have multiple scripts handling individual URLs without mutual interference. These scripts are invoked within another multithreaded architecture. Is that possible?

**Reply by mdmintz** [reply]
Other architectures might work for multithreading, but for the non-officially-supported architectures, it’s up to the individuals to make them work on their own. You’ll need to make sure to do proper thread-locking in order to prevent conflicts when spinning up browsers. The officially supported multithreading architectures do that if you follow those examples with the included instructions.
